tendrl/ceph_integration/objects/rbd/atoms/create/__init__.py

```diff
@@ -1,18 +1,169 @@
+import json
+import uuid
+from sets import Set
+
 from tendrl.ceph_integration.manager.crud import Crud
 from tendrl.ceph_integration.manager.exceptions import \
     RequestStateError
 from tendrl.commons import objects
-from tendrl.ceph_integration.objects.rbd import Rbd
+from tendrl.ceph_integration.objects.pool import Pool
 from tendrl.commons.event import Event
+from tendrl.commons.objects.job import Job
 from tendrl.commons.message import Message
 
 
 class Create(objects.BaseAtom):
-    obj = Rbd
+
     def __init__(self, *args, **kwargs):
         super(Create, self).__init__(*args, **kwargs)
 
     def run(self):
+        Event(
+            Message(
+                priority="info",
+                publisher=NS.publisher_id,
+                payload={
+                    "message": "Checking if a new  pool has to be created or "
+                               "not for rbd creation"
+                },
+                job_id=self.parameters['job_id'],
+                flow_id=self.parameters['flow_id'],
+                cluster_id=NS.tendrl_context.integration_id,
+            )
+        )
+        if not self.parameters.get('Rbd.pool_id'):
+            # Checking if mandatory parameters for pool creation are present
+            mandatory_pool_params = Set(["Rbd.pool_poolname",
+                                         "Rbd.pool_pg_num",
+                                         "Rbd.pool_size",
+                                         "Rbd.pool_min_size"])
+            missing_params = list(mandatory_pool_params.difference(
+                Set(self.parameters.keys())))
+            if not missing_params:
+                # Mapping the passed pool parameters into required keys
+                pool_parameters={}
+                for key, value in self.parameters.items():
+                    if "Rbd.pool_" in key:
+                        pool_parameters[key.replace("Rbd.pool_", "Pool.")] =\
+                            value
+                payload = {
+                    "integration_id": NS.tendrl_context.integration_id,
+                    "run": "ceph.flows.CreatePool",
+                    "status": "new",
+                    "parameters": pool_parameters,
+                    "parent": self.parameters['job_id'],
+                    "type": "sds",
+                    "tags": ["ceph/mon"]
```

Comment:
oops, the tags are wrong on this job

use https://github.com/Tendrl/ceph-integration/blob/develop/tendrl/ceph_integration/objects/definition/ceph.yaml#L5